modules/tinymce/src/plugins/accordion/main/ts/ui/Buttons.ts

```diff
@@ -0,0 +1,25 @@
+import { Menu, Toolbar } from '@ephox/bridge';
+
+import Editor from 'tinymce/core/api/Editor';
+
+const isSummary = (node: Node): node is HTMLElement =>
+  node.nodeName === 'SUMMARY';
+
+const isInSummary = (editor: Editor): boolean => {
+  const node = editor.selection.getNode();
+  return isSummary(node) || Boolean(editor.dom.getParent(node, isSummary));
+};
+
+const onSetup = (editor: Editor) => (buttonApi: Toolbar.ToolbarButtonInstanceApi | Menu.MenuItemInstanceApi) => {
+  const f = () => buttonApi.setEnabled(!isInSummary(editor));
```

Comment:
Maybe describe this one with a name since if we need to have another event listener here later. That is likely if we toggle say content editable false on/off on the body and want to disable the button then the `f` would need to be changed. So better to change it now. I think in general it better to have a describing name unless the function is some random function that is not known say in a compose or map or something.

```suggestion
  const nodeChanged = () => buttonApi.setEnabled(!isInSummary(editor));
```

Reply:
I think I've copied it from somewhere, sorry! Changed the name to `onNodeChange`.